news_app/lib/service/news_service.dart

```diff
@@ -1,11 +1,14 @@
 import 'package:dio/dio.dart';
 import 'package:news_app/model/article_model.dart';
+import 'package:pretty_dio_logger/pretty_dio_logger.dart';
 
 class NewsService {
   final Dio dio;
   final String apiKey = '1c0f731cca954a13875e6965f9c7e9de';
 
-  const NewsService({required this.dio});
+  NewsService({required this.dio}){
+    dio.interceptors.add(PrettyDioLogger());
```

Comment:
add it in the `dioProvider` instead of adding it here

Reply:
this way?

```
final dioProvider = Provider<Dio>(
  (_) {
    final dio = Dio(
      BaseOptions(
        baseUrl: 'https://newsapi.org',
      ),
    );
    dio.interceptors.add(PrettyDioLogger());
    return dio;
  },
);
```

Reply:
Yes

Reply:
Okay, thank you